src/cluster/placement/placements.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package placement
+
+import (
+	"errors"
+	"sort"
+
+	"github.com/m3db/m3/src/cluster/generated/proto/placementpb"
+)
+
+var (
+	errNilPlacementSnapshotsProto = errors.New("nil placement snapshots proto")
+	errEmptyPlacementSnapshots    = errors.New("placement snapshots is empty")
+)
+
+// Placements represents a list of placements that is backward compatible with
+// the deprecated concept of staged placement.
+type Placements []Placement
+
+// NewPlacementsFromProto creates a list of placements from proto.
+func NewPlacementsFromProto(p *placementpb.PlacementSnapshots) (Placements, error) {
+	if p == nil {
+		return nil, errNilPlacementSnapshotsProto
+	}
+
+	placements := make([]Placement, 0, len(p.Snapshots))
+	for _, snapshot := range p.Snapshots {
+		placement, err := NewPlacementFromProto(snapshot)
+		if err != nil {
+			return nil, err
+		}
+		placements = append(placements, placement)
+	}
+
+	// For backward compatibility, we still have to sort
+	// if there are actual staged placements.
+	sort.Sort(placementsByCutoverAsc(placements))
```

Comment:
Just so I understand the semantics--we're still taking the placement with the latest cutover, but we ignore the cutover time when returning a placement (`Last`). Is that right?

Reply:
Oh, good catch - we have to do the same in the `Last`. I missed it in my test because I used this method which already sorts. Let me fix that.

Reply:
Done

Reply:
Can we only keep the snapshot with latest cutover time?

Reply:
> Can we only keep the snapshot with latest cutover time?

+1, dunno why that didn't occur to me haha.

Reply:
great point - done.